src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -129,8 +129,9 @@ impl CommandRunner {
     directory_path: RelativePath,
   ) -> Result<Option<(Tree, Vec<Digest>)>, String> {
     let sub_trie = match root_trie.entry(&directory_path)? {
-      Some(directory::Entry::Directory(d)) => d.tree(),
       None => return Ok(None),
+      Some(directory::Entry::Directory(d)) => d.tree(),
+      Some(directory::Entry::Symlink(_)) => todo!(),
```

Comment:
Error here for same reasons as this errors for files. An output _directory_ was expected but it was not a directory.